app/fruits/[fruitName]/page.js

```diff
@@ -12,31 +14,53 @@ import Fruit from './Fruit';
 // ];
 
 export function generateMetadata({ params }) {
-  const singleFruit = fruits.find((fruit) => {
-    return fruit.name.toLowerCase() === params.fruitName;
-  });
+  const fruit = getFruitByName(params.fruitName);
 
-  if (!singleFruit) {
+  if (!fruit) {
     return rootNotFoundMetadata;
   }
 
   return {
-    title: singleFruit.name,
-    description: `Single fruit page for ${singleFruit.name}`,
+    title: fruit.name,
+    description: `Single fruit page for ${fruit.name}`,
   };
 }
 
 // we add this only if we have no dynamic function as cookies or headers
 export const dynamic = 'force-dynamic';
 
 export default function FruitPage({ params }) {
-  const singleFruit = fruits.find((fruit) => {
-    return fruit.name.toLowerCase() === params.fruitName;
-  });
+  const fruit = getFruitByName(params.fruitName);
 
-  if (!singleFruit) {
+  if (!fruit) {
     notFound();
   }
 
-  return <Fruit fruit={singleFruit} />;
+  const fruitComments = parseJson(getCookie('fruitComments'));
+
+  const currentComments = Array.isArray(fruitComments)
+    ? fruitComments
+    : [];
+
+  const fruitComment = currentComments.find(
+    (singleFruitComment) => singleFruitComment.id === fruit.id,
+  );
+
+  return (
+    <>
+      <h1>{params.fruitName}</h1>
+      <span
```

Comment:
let's use a `div` here instead of `span`

Reply:
5dcbd9c

Reply:
5dcbd9c

Reply:
[ccaf7ec](https://github.com/upleveled/next-js-example-winter-2023-vienna-austria/pull/15/commits/ccaf7ec58f04066b612c7338707027caec1386ae)

Reply:
[9cc48ab](https://github.com/upleveled/next-js-example-winter-2023-vienna-austria/pull/15/commits/9cc48abce62cad03c31a8a986da09becea59c704)